adafruit_displayio_layout/widgets/widget.py

```diff
@@ -0,0 +1,164 @@
+# SPDX-FileCopyrightText: 2021 Kevin Matocha
+#
+# SPDX-License-Identifier: MIT
+"""
+
+`widget`
+================================================================================
+CircuitPython GUI Widget Class for visual elements
+
+* Author(s): Kevin Matocha
+
+Implementation Notes
+--------------------
+
+**Hardware:**
+
+**Software and Dependencies:**
+
+* Adafruit CircuitPython firmware for the supported boards:
+  https://github.com/adafruit/circuitpython/releases
+
+"""
+
+import displayio
+
+__version__ = "0.0.0-auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_DisplayIO_Layout.git"
+
+
+class Widget(displayio.Group):
+    """
+    A Widget class definition for graphical display elements.
+
+    :param int x: pixel position
+    :param int y: pixel position
+    :param int width: width of the switch in pixels, set to ``None`` to auto-size
+     relative to the height
+    :param int height: height of the switch in pixels
+    :param str name: name of the switch
+    :param float anchor_point: (X,Y) values from 0.0 to 1.0 to define the anchor
+     point relative to the switch bounding box
+    :param int anchored_position: (x,y) pixel value for the location
+     of the anchor_point
+
+    """
+
+    def __init__(
+        self,
+        width=None,
```

Comment:
In the docstring above `x` and `y` are listed as the first two arguments, but they aren't here inside of __init__ parameters.

Is that because they get brought in with **kwargs below? If so I wonder if it would break anything if we added them explicitly I think for the sake of reading the code and understanding it would be nice if they were listed here if possible.

Reply:
Thanks for the suggestion, I struggled with whether I should explicitly call them out here or if someone should have to go up the class tree to find them.  

I agree with you to include x,y in the `__init__` definition. So, I added them back in here.  

Also I added scale, along with a warning, that if scale > 1 that you have to take care of scaling the `touch_boundary`.